### PR TITLE
Fix CI builds on Windows and Linux

### PR DESCRIPTION
**Pull request by RytoEX**
This PR will (hopefully) fix CI builds on Windows and Linux.  This addresses items 1, 2, and partly 3 from #143.  I figured I'd work on items 3 and (optionally) 4 if this PR works, and if those items are wanted.  This will need to be merged before the CI services will use the updated config files.

Commits are squashable if needed/desired.  Feedback and reviews welcome.

### CODE REVIEW

**Comment by haganbmj** [comment]
Appveyor is using the updated config it seems.
I know travis will just restrict access to sensitive environment variables, so maybe appveyor does something similar.

I've got it building over on my fork as backup just in case though.
https://github.com/haganbmj/obs-websocket/commits/fix-ci
https://ci.appveyor.com/project/haganbmj/obs-websocket/build/1.0.111

File: appveyor.yml [hunk 1]
```diff
@@ -6,25 +6,27 @@ install:
...
-  - if not exist qt570.zip curl -kLO https://www.slepin.fr/obs-websocket/ci/qt570.zip -f --retry 5 -C -
-  - 7z x qt570.zip -o"Qt5.7.0"
   - set DepsPath32=%CD%\dependencies2013\win32
   - set DepsPath64=%CD%\dependencies2013\win64
-  - set QTDIR32=%CD%\Qt5.7.0\msvc2013
-  - set QTDIR64=%CD%\Qt5.7.0\msvc2013_64
+  - set QTDIR32=C:\Qt\5.7\msvc2013
+  - set QTDIR64=C:\Qt\5.7\msvc2013_64
   - set build_config=Release
   - git clone --recursive https://github.com/jp9000/obs-studio
```
Comment (haganbmj) [inline appveyor.yml]: If the `lib`s are stored is there a need to checkout `obs-studio` and run cmake?

Comment (RytoEX) [reply appveyor.yml]: I'm not entirely sure.  The CMake file for obs-websocket uses LIBOBS_INCLUDE_DIR.  I figured that if it wasn't available and specified that CMake might fail.  I could always test it later, but I was trying to make minimal changes at first.

**Comment by RytoEX** [comment]
Hmm.  It seems that the caching for the .lib files didn't work because %build_config% wasn't expanded.  I'll have to fix that.

**Comment by RytoEX** [comment]
I'm testing on a different branch just to keep from repeatedly hitting this PR with notifications.  I've fixed most of the caching issues there.  I'm still cleaning it up and testing some more to try to prevent failures.  I'll update the branch for this PR once I'm satisfied with my fixes.

**Comment by Palakis** [comment]
Here are the links for recompressed Qt 5.7.0:
- Full: https://www.slepin.fr/obs-plugins/deps/qt570.7z
- QtNetwork and QtWebSockets, binaries only: http://www.slepin.fr/obs-plugins/deps/qt570_bin_network+websockets.7z

**Comment by RytoEX** [comment]
Thanks!  I'll get to this later tonight or this weekend.

**Comment by RytoEX** [comment]
Building for the first time now takes about 10-11 minutes (the Qt 5.7.0 is half the size, so it takes less time to download).  Building after an obs-studio build is cached now takes about 3-4 minutes.  I'll push the changes here soon.  I have some thoughts on how to further improve CI build times, but I think I'll address those in a separate PR.

It should be fine to continue using the full Qt 5.7.0 for compiling until OBS switches to a newer Qt version unless your AppVeyor cache is too full because of other projects on your account.

**Comment by Palakis** [comment]
Looks good to me. Thanks @RytoEX and @haganbmj!

**Comment by RytoEX** [comment]
Looks like the cache is still rejecting uploads for now:

> Updating build cache...
Cache 'C:\projects\dependencies2013.zip' - Up to date
Cache 'C:\projects\qt570.7z' - Uploading (451,532,769 bytes)...100%Error uploading cache entry to the cache storage: Remote server returned 500: There is not enough space on the disk.
Cache 'C:\projects\obs-studio-last-tag-built.txt' - Uploading (199 bytes)...100%Error uploading cache entry to the cache storage: Remote server returned 500: There is not enough space on the disk.
Cache 'C:\projects\obs-studio' - Uploading (238,703,652 bytes)...100%Error uploading cache entry to the cache storage: Remote server returned 500: There is not enough space on the disk.
Cache 'C:\projects\qt570.zip' - Deleted

That might have been because qt570.zip hadn't yet been deleted.  We'll have to monitor the next few builds to see how it behaves.

**Pull request merged**